compiler/semtypes.nim

```diff
@@ -1396,7 +1396,10 @@ proc semTypeNode(c: PContext, n: PNode, prev: PType): PType =
         fixupTypeOf(c, prev, typExpr)
         result = typExpr.typ
       else:
-        result = semTypeExpr(c, n, prev)
+        if c.inGenericContext > 0 and n.kind == nkCall:
```

Comment:
why is only ``nkCall`` checked here and not all call kinds?

Reply:
This fix turned out to be a bit tricky, due to some tricks involving the use of types and operators (as in the `T -> U` syntax from module for example). That's why I avoid the handling of such operators here.

All in all, I'm not particularly happy with this broad application of `tyFromExpr` and my first attempt was to try to introduce more logic in the compiler allowing unresolved statics to pass-through sem without executing the macros where they are used as arguments, but this turned out to be quite involved. We still need a lot of work on enabling implicit statics by default and I think these developments will allow for a nicer solution in the future.